draft-ietf-mls-protocol.md

```diff
@@ -4836,6 +4846,53 @@ be securely removed from a group. It also allows a member to rotate their
 keypair such that the old private key can no longer be used to decrypt new
 messages.
 
+## Confidentiality of Group Metadata
+
+MLS handles certain metadata about a group that is sometimes exposed to parties
+outside the group:
+
+* Group ID
+* Group membership
+* Group extensions
+* Frequency of epoch changes
+* Frequency of application messages within an epoch
+
+This information is typically protected from parties other than the DS by
+applying "hop by hop" transport encryption on communications with the DS (in
+contrast to the "end to end" protections provided by MLS). The protection this
+information receives with regard to the DS depends on how MLS is operated.
+Overall, MLS does not provide robust protections against the DS observing the
+group's metadata.
+
+MLS provides no mechanism to protect the group ID and epoch of a message from
+the DS, so group ID and the frequency of messages and epoch changes are not
+protected against inspection by the DS.
+
+If a group can be joined by external commits (see
+{{joining-via-external-commits}}), then a group member will need to publish a
+GroupInfo object that states the group's extensions and a commitment to the
+group's membership.  This generally results in the group's membership and
+extensions being known to the DS and prospective new joiners.
```

Comment:
No necessarily. The GroupInfo could be uploaded in encrypted form and then the key material could be passed to a prospective joiner.

It might also be worth mentioning here that the same goes if the tree is not sent as an extension via the Welcome.